### DISCUSSION TITLE
Update status from software

### DISCUSSION THREAD

**Discussion by lpuntaloro** [HomeSpan/HomeSpan#371] [Q&A]
Hi.. I have a question. I would like to implement homespan in my self-produced esp32-based home automation. I would like when I turn on a light at the push of a button, I would also update Apple's home status related to that light. Is that possible? How can I do that? 
Thank you

**Comment by HomeSpan** [comment]
Yes, please see [Tutorial Example 15 - RealPushButtons](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Tutorials.md#example-15---realpushbuttons)

**Reply by lpuntaloro** [reply]
Hi, thank you for your reply. I saw example 15 and it works a little differently than I would like .. I would like not to use the class implemented by homespan for buttons because I need to do other operations that are already in use .. is it possible to update the status of the lamp on home by inserting something like update into existing routines?
Thanks

**Comment by HomeSpan** [comment]
Yes, `SpanButton()` only detects presses, it does not actually update anything.  The `setVal()` method for Characteristics is what actually does the updating of the Home App by relaying new information.  You can see the use of this method in the `button()` section of the example, but it can be used on a standalone basis.  

For a specific tutorial on `setVal()`, see [Example 12 - ServiceLoops](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Tutorials.md#example-12---serviceloops).  Note `setVal()` can be used anywhere, not just inside Service Loops.  However, it is within this example that `setVal()` is explained.

**Reply by lpuntaloro** [reply]
Thank you for your reply, I'll try.. if I have any difficulties can I contact you? Thank you

**Reply by HomeSpan** [reply]
Sure thing - let me know if you have any issues/questions.
